tmva/tmva/src/RBDT.cxx

```diff
@@ -1,4 +1,389 @@
-#include "TMVA/RBDT.hxx"
+/**********************************************************************************
+ * Project: ROOT - a Root-integrated toolkit for multivariate data analysis       *
+ * Package: TMVA                                                                  *
+ *                                                                                *
+ *                                                                                *
+ * Description:                                                                   *
+ *                                                                                *
+ * Authors:                                                                       *
+ *      Jonas Rembser (stefan.wunsch@cern.ch)                                     *
+ *                                                                                *
+ * Copyright (c) 2024:                                                            *
+ *      CERN, Switzerland                                                         *
+ *                                                                                *
+ * Redistribution and use in source and binary forms, with or without             *
+ * modification, are permitted according to the terms listed in LICENSE           *
+ * (see tmva/doc/LICENSE)                                          *
+ **********************************************************************************/
 
-template class TMVA::Experimental::RBDT<TMVA::Experimental::BranchlessForest<float>>;
-template class TMVA::Experimental::RBDT<TMVA::Experimental::BranchlessJittedForest<float>>;
+#include <TMVA/RBDT.hxx>
+
+#include <TFile.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <sstream>
+#include <stdexcept>
+#include <stdlib.h>
+#include <string>
+#include <vector>
+
+namespace {
+
+namespace detail {
+
+typedef std::map<int, int> IndexMap;
+
+void correctIndices(std::vector<int>::iterator begin, std::vector<int>::iterator end, IndexMap const &nodeIndices,
+                    IndexMap const &leafIndices)
+{
+   for (std::vector<int>::iterator it = begin; it != end; ++it) {
+      if (nodeIndices.count(*it)) {
+         *it = nodeIndices.at(*it);
+      } else if (leafIndices.count(*it)) {
+         *it = -leafIndices.at(*it);
+      } else {
+         throw std::runtime_error("something is wrong in the node structure");
+      }
+   }
+}
+
+template<class Value_t>
+void softmaxTransformInplace(Value_t *out, int nOut)
```

Comment:
I would like to keep the C-style interface for compatibility reasons with the original fastforest, in case I want to sync code again.

There is just no alternative to it yet. `std::array` and `std::vector` are too specific, and `std::span` only comes in C++20.